pkg/setting/setting_smtp.go

```diff
@@ -38,4 +44,34 @@ func (cfg *Cfg) readSmtpSettings() {
 	cfg.Smtp.SendWelcomeEmailOnSignUp = emails.Key("welcome_email_on_sign_up").MustBool(false)
 	cfg.Smtp.TemplatesPatterns = util.SplitString(emails.Key("templates_pattern").MustString("emails/*.html, emails/*.txt"))
 	cfg.Smtp.ContentTypes = util.SplitString(emails.Key("content_types").MustString("text/html"))
+
+	// populate static headers
+	if err := cfg.readGrafanaSmtpStaticHeaders(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (cfg *Cfg) readGrafanaSmtpStaticHeaders() error {
+	staticHeadersSection := cfg.Raw.Section("smtp.static_headers")
+	keys := staticHeadersSection.Keys()
+	cfg.Smtp.StaticHeaders = make(map[string]string, len(keys))
+
+	for _, key := range keys {
+		labelName := model.LabelName(key.Name())
+		labelValue := model.LabelValue(key.Value())
+
+		if !labelName.IsValid() {
```

Comment:
Is there an SMTP-specific validation we could use? For example, dots are allowed in SMTP headers, but not in prometheus labels.

From a search it looks like SMTP uses standard MIME headers (though I am not an expert in this area, so please double check), and there's a tool in the stdlib for dealing with these: https://pkg.go.dev/mime#WordDecoder

To validate, we might be able to just encode and decode the given header, and pass if it works with no errors.

Reply:
Sure, I'll add some regex to check for that